### PR TITLE
Search feature

### PR DESCRIPTION
**Pull request by bradTpeters**
Patch v2, incorporates feedback by William Douglas

### CODE REVIEW

**Comment by bradTpeters** [comment]
New v3 pushed, should incorporate all feedback. Thanks all!

File: src/search.c [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+
+int search_main(int argc, char **argv)
+{
+	int ret = 0;
+	int lock_fd;
+	struct manifest *MoM = NULL;
+
+	if (!parse_options(argc, argv) ||
+		create_required_dirs()) {
+		free_globals();
```
Comment (phmccarty) [inline src/search.c]: init_globals() is called the first time a few lines down, so you'll want to avoid calling it here.

File: src/swupd.c [hunk 2]
```diff
@@ -39,6 +39,7 @@ static struct subcmd commands[] = {
 	{ "update", "Update to latest OS version", update_main },
 	{ "verify", "Verify content for OS version", verify_main },
 	{ "check-update", "Checks if a new OS version is available", check_update_main},
+	{ "search", "Search Clear Linux for a binary or library", search_main },
```
Comment (phmccarty) [inline src/swupd.c]: This command does not appear to be specific to Clear Linux. Do we need to mention Clear Linux here?

File: src/search.c [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+	char *tarfile, *untard_file, *url;
+
+	while (list) {
+		file = list->data;
+		list = list->next;
+
+		string_or_die(&untard_file, "%s/%i/Manifest.%s", STATE_DIR, file->last_change,
+						file->filename);
+
+		string_or_die(&tarfile, "%s/%i/Manifest.%s.tar", STATE_DIR, file->last_change,
```
Comment (phmccarty) [inline src/search.c]: The tarfile string allocated here is unused in this function, and the memory is not freed.

**Comment by bradTpeters** [comment]
Archana, lmk when your back in. We should discuss the load_manifests() question, and whether a reversing of the arguments order makes sense

File: src/swupd.c (hunk 2 above)
Comment (bradTpeters) [inline src/swupd.c]: The search is specifically over the manifests/bundles provided by Clear Linux, so I thought it made sense.

**Comment by bradTpeters** [comment]
And one more

**Comment by bradTpeters** [comment]
Cleanup from prev patch added.

File: src/swupd.c (hunk 2 above)
Comment (phmccarty) [inline src/swupd.c]: On second thought, I'm okay with leaving this for now.

File: src/search.c (hunk 3 above)
Comment (phmccarty) [inline src/search.c]: Comment addressed :-)

File: src/swupd.c (hunk 2 above)
Comment (bradTpeters) [inline src/swupd.c]: Great

**Comment by phmccarty** [comment]
Looks good to me. We can iterate from this point. I did a squash/merge to master.

**Pull request closed without merging**